core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -136,33 +137,42 @@ public ResponsePath doWork(PointList waypoints, List<Path> paths, EncodedValueLo
 
                 fullPoints.add(tmpPoints);
                 responsePath.addPathDetails(PathDetailsFromEdges.calcDetails(path, evLookup, weighting, requestedPathDetails, pathBuilderFactory, origPoints, graph));
+                wayPointIntervals.add(new ResponsePath.Interval(origPoints, pathIndex < paths.size() - 1 ? fullPoints.size() : fullPoints.size() - 1));
                 origPoints = fullPoints.size();
             }
 
             allFound = allFound && path.isFound();
         }
 
-        if (!fullPoints.isEmpty()) {
-            responsePath.addDebugInfo("simplify (" + origPoints + "->" + fullPoints.size() + ")");
-            if (fullPoints.is3D)
-                calcAscendDescend(responsePath, fullPoints);
-        }
+        if (!fullPoints.isEmpty() && fullPoints.is3D)
+            calcAscendDescend(responsePath, fullPoints);
 
         if (enableInstructions) {
             fullInstructions = updateInstructionsWithContext(fullInstructions);
             responsePath.setInstructions(fullInstructions);
         }
 
         if (!allFound) {
-            responsePath.addError(new ConnectionNotFoundException("Connection between locations not found", Collections.<String, Object>emptyMap()));
+            responsePath.addError(new ConnectionNotFoundException("Connection between locations not found", Collections.emptyMap()));
+        }
+
+        if (allFound) {
```

Comment:
Can we not just return the responsePath with the `connection-not-found` error as soon as one of the connections isn't found, or why do we keep going afterwards?

Reply:
The idea is that we get a full list of all errors, not just the one for the first leg that fails. Will leave this as it is since it is rather unrelated.